src/util.rs

```diff
@@ -136,88 +136,120 @@ pub fn respace(s: &str) -> String {
     s.split_whitespace().collect::<Vec<_>>().join(" ")
 }
 
-pub struct ExpandedRegister<'a> {
-    pub info: &'a RegisterInfo,
-    pub name: String,
-    pub offset: u32,
-    pub ty: Either<String, Rc<String>>,
+pub fn register_to_rust(register: &Register) -> Tokens {
+    match *register {
+        Register::Single(ref _info) => {
+            let comment = &format!(
+                "0x{:02x} - {}",
+                register.address_offset,
+                respace(&register.description)
+            )
+                [..];
+
+            let array_index_match = Regex::new(r"%s\((?P<i>[ a-zA-Z0-9]*)\)").unwrap();
+            //println!("{}", register.name.as_str());
+            let rty = Ident::new(array_index_match.replace_all(register.name.as_str(), "").to_sanitized_upper_case());
+            let reg_name = Ident::new(array_index_match.replace_all(register.name.as_str(), "$i").to_sanitized_snake_case());
```

Comment:
OK. I see why "%s(N)" is being used: to have expanded registers use the same type. This ... feels hackish; I would prefer if this was encoded in the type system rather than using a regex replace. ExpandedRegisters was used for this reason (to make all the expanded registers have the same type).